diff --git a/support/scripts/br2-external b/support/scripts/br2-external
index 8aea479d20..d9b970de3a 100755
--- a/support/scripts/br2-external
+++ b/support/scripts/br2-external
@@ -34,6 +34,14 @@ main() {
     trap "error 'unexpected error while generating ${ofile}\n'" ERR
 
     mkdir -p "${outputdir}"
+
+    if [ ! -f "${outputdir}/.pbuilder-br2-external-already-executed" ]; then
+        echo "Lock file doesn't exist, execute normally" >> ${outputdir}/.pbuilder-lockfile-debug
+    else
+        echo "Lock file already exist, return" >> ${outputdir}/.pbuilder-lockfile-debug
+        return
+    fi
+
     do_validate "${outputdir}" ${@//:/ }
     do_mk "${outputdir}"
     do_kconfig "${outputdir}"
