diff --git a/Makefile b/Makefile
index 05b790d770..0d9a4bede9 100644
--- a/Makefile
+++ b/Makefile
@@ -919,6 +919,12 @@ show-info:
 		) \
 	)
 
+PHONY: pbuilder
+pbuilder:
+	@cd $(BUILD_DIR)/..; \
+	$(TOPDIR)/support/scripts/pbuilder.py -l 2 $(BR2_CONFIG) $(TOPDIR) $(BR2_CONFIG) $(TOPDIR)/utils/pbuilder; \
+	TOPDIR=$(TOPDIR) CONFIG_DIR=$(CONFIG_DIR) $(TOPDIR)/utils/pbuilder/src/pbuilder -l 1 -f $(CONFIG_DIR)/.pbuilder.deps
+
 .PHONY: pkg-stats
 pkg-stats:
 	@cd "$(CONFIG_DIR)" ; \
