diff --git a/Makefile b/Makefile
index 36896684b1..6c340dfe8b 100644
--- a/Makefile
+++ b/Makefile
@@ -918,6 +918,12 @@ show-info:
 		) \
 	)
 
+.PHONY: pbuilder
+pbuilder:
+	@cd $(CONFIG_DIR); \
+	$(TOPDIR)/support/scripts/pbuilder.py -l 2 $(BR2_CONFIG) $(BUILD_DIR) $(TOPDIR)/utils/pbuilder; \
+	TOPDIR=$(TOPDIR) CONFIG_DIR=$(CONFIG_DIR) BR2_EXTERNAL=$(BR2_EXTERNAL) $(TOPDIR)/utils/pbuilder/src/pbuilder -l 1 -f .pbuilder.deps
+
 .PHONY: pkg-stats
 pkg-stats:
 	@cd "$(CONFIG_DIR)" ; \
@@ -1099,6 +1105,7 @@ show-vars:
 clean:
 	rm -rf $(BASE_TARGET_DIR) $(BINARIES_DIR) $(HOST_DIR) $(HOST_DIR_SYMLINK) \
 		$(BUILD_DIR) $(BASE_DIR)/staging \
+		$(BASE_DIR)/pbuilder_logs $(BASE_DIR)/.pbuilder.deps \
 		$(LEGAL_INFO_DIR) $(GRAPHS_DIR) $(PER_PACKAGE_DIR) $(CPE_UPDATES_DIR) \
 		$(O)/pkg-stats.*
 
