diff --git a/Makefile b/Makefile
index 6d5e9a7972..fc89cad4e5 100644
--- a/Makefile
+++ b/Makefile
@@ -924,6 +924,13 @@ show-info:
 		) \
 	)
 
+.PHONY: pbuilder
+pbuilder:
+	@cd $(CONFIG_DIR); \
+	$(TOPDIR)/support/scripts/pbuilder.py $(BR2_CONFIG) $(BUILD_DIR) $(TOPDIR)/utils/pbuilder || \
+		{ echo "Script $(TOPDIR)/support/scripts/pbuilder.py failed!"; exit 1; }; \
+	CONFIG_DIR=$(CONFIG_DIR) BR2_EXTERNAL=$(BR2_EXTERNAL) BUILD_DIR=$(BUILD_DIR) $(TOPDIR)/utils/pbuilder/src/pbuilder -f .pbuilder.deps
+
 .PHONY: pkg-stats
 pkg-stats:
 	@cd "$(CONFIG_DIR)" ; \
@@ -1102,6 +1109,7 @@ show-vars:
 clean:
 	rm -rf $(BASE_TARGET_DIR) $(BINARIES_DIR) $(HOST_DIR) $(HOST_DIR_SYMLINK) \
 		$(BUILD_DIR) $(BASE_DIR)/staging \
+		$(BASE_DIR)/pbuilder_logs $(BASE_DIR)/.pbuilder* \
 		$(LEGAL_INFO_DIR) $(GRAPHS_DIR) $(PER_PACKAGE_DIR) $(O)/pkg-stats.*
 
 .PHONY: distclean
